engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/FieldCascadable.java

```diff
@@ -85,14 +84,9 @@ public FieldCascadable build() {
 		}
 
 		/**
-		 * Returns an accessible version of the given member. Will be the given member itself in case it is accessible,
-		 * otherwise a copy which is set accessible.
+		 * Returns an accessible copy of the given member.
 		 */
 		private Field getAccessible(Field original) {
-			if ( ( (AccessibleObject) original ).isAccessible() ) {
```

Comment:
Hum, hum, a pity to see that go. Isn't there some new way of doing this, e.g. via `canAccess()`? It's a generally interesting question for our projects.

Reply:
As explained in the commit message, this call is not useful. Keep in mind that `isAccessible()` does not check if the method is accessible. It checks that the security permissions have previously been overridden using `setAccessible()`. That's a common misconception.

It's not the first time I'm tempted to remove these calls as they are useless, considering we call `setAccessible()` only once per element and we do that on a copy anyway.

As for calling `canAccess()`, I started to go the MR jar way and try to see how we could do that but, AFAICS, to use `canAccess()` on a method or a field, you need an instance of the object, which we don't have in this part of the code.

At some point, we might want to take that path but I'm pretty sure it's not going to be pretty and it's an orthogonal issue.

This commit only removes a useless call.

Reply:
Just wanted to add that most likely this would go away later, when we switch to using handles. I remember when I was experimenting with the handles I also removed this `isAccessible` check, as it was causing problems.